File: uber-backend/uber-backend/src/main/java/com/example/uberbackend/model/Message.java
package com.example.uberbackend.model;

import com.example.uberbackend.model.enums.MessageStatus;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor @AllArgsConstructor
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private long id;
    private String senderEmail;
    private String receiverEmail;
    private String content;
    private LocalDateTime date;
    private MessageStatus status;

    public Message(String sender, String receiver, String content, LocalDateTime date, MessageStatus status) {
        this.senderEmail = sender;
        this.receiverEmail = receiver;
        this.content = content;
        this.date = date;
        this.status = status;
    }
}

File: uber-frontend/src/app/model/Message.ts
export interface Message{
    senderEmail: string,
    senderFirstName: string | null,
    senderLastName: string | null,
    senderImage: string | null,
    receiverEmail: string,
    receiverFirstName: string | null,
    receiverLastName: string | null,
    receiverImage: string | null,
    content: string,
    date: Date,
    status: number
  }